string               topic
string               datatype
TopicRemoteDetails[] remotes